### PR TITLE
Squash ReadOnlyLedger hierarchy [DPP-809]

### PR DESCRIPTION
**Pull request by tudor-da**
changelog_begin
changelog_end

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by nmarton-da — COMMENTED** [review]
Definitely goes in the right direction! :raised_hands: 
Added some comments on the task, let me know if you would like to accommodate some in this PR and I should review later, or keep it incremental and you need a review now

**Comment by tudor-da** [comment]
> Definitely goes in the right direction! 🙌 Added some comments on the task, let me know if you would like to accommodate some in this PR and I should review later, or keep it incremental and you need a review now

Cool. Let's do it in this task.

File: ledger/participant-integration-api/src/main/scala/platform/index/ReadOnlyLedgerImpl.scala [hunk 1]
```diff
@@ -30,12 +31,11 @@ import com.daml.platform.PruneBuffers
 import com.daml.platform.akkastreams.dispatcher.Dispatcher
 import com.daml.platform.akkastreams.dispatcher.SubSource.RangeSource
 import com.daml.platform.store.appendonlydao.{LedgerDaoTransactionsReader, LedgerReadDao}
-import com.daml.ledger.participant.state.index.v2.MeteringStore.TransactionMetering
 import com.daml.platform.store.entries.{ConfigurationEntry, PackageLedgerEntry, PartyLedgerEntry}
 
 import scala.concurrent.Future
 
-private[platform] abstract class BaseLedger(
+private[index] class ReadOnlyLedgerImpl(
```
Comment (nmarton-da) [inline ledger/participant-integration-api/src/main/scala/platform/index/ReadOnlyLedgerImpl.scala:38]: so nice, so we see now ReadOnlySqlLedger was really a factory for BaseLedger!

**Review by nmarton-da — APPROVED** [review]
Looks very good to me!
Left some improvement comments.

**Pull request merged**